### PR TITLE
Simplify and harmonize grub menu style

### PR DESCRIPTION
**Pull request by didrocks**
Try to simplify and harmonize our shell style in grub menu generation
script.

### CODE REVIEW

File: contrib/grub/15_linux_zfs [hunk 1]
```diff
@@ -638,38 +638,38 @@ zfs_linux_entry () {
...
-        echo "	gfxmode \$linux_gfx_mode" | sed "s/^/$submenu_indentation/"
+    if ([ "${ubuntu_recovery}" = 0 ] || [ "${type}" != "recovery" ]) && \
+        ([ "${GRUB_GFXPAYLOAD_LINUX}" != "" ] || [ "${gfxpayload_dynamic}" = 1 ]); then
+        echo "	gfxmode \${linux_gfx_mode}" | sed "s/^/${submenu_indentation}/"
     fi
 
-    echo "	insmod gzio" | sed "s/^/$submenu_indentation/"
-    echo "	if [ x\$grub_platform = xxen ]; then insmod xzio; insmod lzopio; fi" | sed "s/^/$submenu_indentation/"
+    echo "	insmod gzio" | sed "s/^/${submenu_indentation}/"
+    echo "	if [ \${grub_platform} = xen ]; then insmod xzio; insmod lzopio; fi" | sed "s/^/${submenu_indentation}/"
```
Comment (jibel) [inline contrib/grub/15_linux_zfs]: It'll break if for whatever reason $grub_platform is empty

**Review by jibel — CHANGES_REQUESTED** [review]
Thanks for the MP. 1 small robustness fix.

File: contrib/grub/15_linux_zfs (hunk 1 above)
Comment (didrocks) [reply contrib/grub/15_linux_zfs]: Thanks! Fixed in new commit.

**Review by jibel — APPROVED** [review]
LGTM, Thanks !

**Pull request merged**